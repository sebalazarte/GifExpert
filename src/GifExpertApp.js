import React, { Fragment, useState } from 'react';
import { AddCategory } from './components/AddCategory';
import { GifGrid } from './components/GifGrid';

const GifExpertApp = ({defaultCategories = []}) => {

    const [categorias, setCategorias] = useState(defaultCategories);
    const [limite, setLimite] = useState(10);

    return (
        <Fragment>
            <h1>Gif Expert App</h1>
            <hr />
            <AddCategory setCategorias={setCategorias} 
                         setLimite={setLimite} />
            <ol>
                {
                    categorias.map(categoria => (
                        // return (<li key={categoria}> {categoria} </li>);
                         <GifGrid 
                            key={categoria}
                            category={categoria}
                            limit={limite}
                            />
                    ))
                }
            </ol>

        </Fragment>);

}

export default GifExpertApp



