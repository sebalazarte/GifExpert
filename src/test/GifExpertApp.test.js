import React from 'react';
import { shallow } from 'enzyme';
import '@testing-library/jest-dom';
import GifExpertApp from '../GifExpertApp';

describe('Pruebas con el componente principal', () => {
    
    test('Debe mostrarse correctamente', () => {
        
        const wrapper = shallow(<GifExpertApp/>);
        expect(wrapper).toMatchSnapshot();
    });

    test('Debe de mostrar una lista de categorias', () => {
        const categories = ['One punch', 'Dragon ball'];
        const wrapper = shallow(<GifExpertApp defaultCategories={categories}/>);
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('GifGrid').length).toBe(categories.length);
    });

});