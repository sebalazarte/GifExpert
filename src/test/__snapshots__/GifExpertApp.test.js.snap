// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas con el componente principal Debe de mostrar una lista de categorias 1`] = `
<Fragment>
  <h1>
    Gif Expert App
  </h1>
  <hr />
  <AddCategory
    setCategorias={[Function]}
    setLimite={[Function]}
  />
  <ol>
    <GifGrid
      category="One punch"
      key="One punch"
      limit={20}
    />
    <GifGrid
      category="Dragon ball"
      key="Dragon ball"
      limit={20}
    />
  </ol>
</Fragment>
`;

exports[`Pruebas con el componente principal Debe mostrarse correctamente 1`] = `
<Fragment>
  <h1>
    Gif Expert App
  </h1>
  <hr />
  <AddCategory
    setCategorias={[Function]}
    setLimite={[Function]}
  />
  <ol />
</Fragment>
`;
